src/OMODFrameworkWrapper.cpp

```diff
@@ -412,3 +416,16 @@ void OMODFrameworkWrapper::displayReadmeSlot(const QString& modName, const QStri
     readmePopup->setAttribute(Qt::WA_DeleteOnClose);
   }
 }
+
+void OMODFrameworkWrapper::showWaitDialogSlot(QString message) {
+  mWaitDialog = new QProgressDialog(message, tr("Cancel"), 0, 0, mParentWidget);
+  mWaitDialog->setWindowFlags(mWaitDialog->windowFlags() & ~Qt::WindowContextHelpButtonHint & ~Qt::WindowCloseButtonHint);
+  mWaitDialog->setWindowModality(Qt::WindowModal);
+  mWaitDialog->setCancelButton(nullptr);
+  mWaitDialog->show();
+}
+
+void OMODFrameworkWrapper::hideWaitDialogSlot() {
+  mWaitDialog->hide();
+  mWaitDialog->deleteLater();
```

Comment:
I'm not super thrilled about there being a function you have to remember to call to ensure the wait dialogue gets deleted. Maybe make the class member a lambda-ish unique_ptr that gets `reset()` here that'll *definitely* get cleared at some point if it's forgotten about or `showWaitDialog` and `hideWaitDialog` calls are mismatched and a nullptr check before calling `hide()`. I'm aware it isn't broken as-is, but it feels like leaving landmines lying in a field just because we're not using the field for anything else right now.

Reply:
I've used a `unique_ptr` with a custom deleter as a `OMODFrameworkWrapper` member.